File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(num):
	nums = range(1, num + 1)
	return _square(sum(nums))

def sum_of_squares(num):
	nums = range(1, num + 1)
	return sum(map(_square, nums))

def difference(num):
	a = square_of_sum(num)
	b = sum_of_squares(num)
	return a - b

def _square(x):
	return x ** 2

File: javascript/difference-of-squares/difference-of-squares.js
const range = n => [...Array(n + 1).keys()].slice(1);
const square = n => n * n;
const sum = (a, b) => a + b;

class Squares {

    constructor(n) {
        let numbers = range(n);
        this.squareOfSums = square(numbers.reduce(sum));
        this.sumOfSquares = numbers.map(square).reduce(sum);
        this.difference = this.squareOfSums - this.sumOfSquares;
    }

}

module.exports = Squares;